test_haystack/solr_tests/server/start-solr-test-server.sh

```diff
@@ -24,35 +25,59 @@ if [ ! -f ${SOLR_ARCHIVE} ]; then
-tar -C solr4 -xf ${SOLR_ARCHIVE} --strip-components 2 solr-${SOLR_VERSION}/example
-tar -C solr4 -xf ${SOLR_ARCHIVE} --strip-components 1 solr-${SOLR_VERSION}/dist solr-${SOLR_VERSION}/contrib
+if [ -z ${SOLR_CORENAME}  ]; then
+    SOLR_CORENAME="collection1"
+fi
+
+if [[ ${SOLR_VERSION} == 4.* ]]; then
+    echo "Extracting Solr ${SOLR_VERSION} to `pwd`/solr4/"
+    rm -rf solr4
+    mkdir solr4
+    tar -C solr4 -xf ${SOLR_ARCHIVE} --strip-components 2 solr-${SOLR_VERSION}/example
+    tar -C solr4 -xf ${SOLR_ARCHIVE} --strip-components 1 solr-${SOLR_VERSION}/dist solr-${SOLR_VERSION}/contrib
 
-echo "Changing into solr4"
+    echo "cd into solr 4 directory"
 
-cd solr4
+    cd solr4
 
-echo "Configuring Solr"
+    echo "Configuring a Solr 4 Core named ${SOLR_CORENAME}"
+    cp ${TEST_ROOT}/solrconfig.xml solr/${SOLR_CORENAME}/conf/solrconfig.xml
+    cp ${TEST_ROOT}/schema.xml solr/${SOLR_CORENAME}/conf/schema.xml
 
-cp ${TEST_ROOT}/solrconfig.xml solr/collection1/conf/solrconfig.xml
-cp ${TEST_ROOT}/schema.xml solr/collection1/conf/schema.xml
+    # Fix paths for the content extraction handler:
+    perl -p -i -e 's|<lib dir="../../../contrib/|<lib dir="../../contrib/|'g solr/*/conf/solrconfig.xml
+    perl -p -i -e 's|<lib dir="../../../dist/|<lib dir="../../dist/|'g solr/*/conf/solrconfig.xml
 
-# Fix paths for the content extraction handler:
-perl -p -i -e 's|<lib dir="../../../contrib/|<lib dir="../../contrib/|'g solr/*/conf/solrconfig.xml
-perl -p -i -e 's|<lib dir="../../../dist/|<lib dir="../../dist/|'g solr/*/conf/solrconfig.xml
+    # Add MoreLikeThis handler
+    perl -p -i -e 's|<!-- A Robust Example|<!-- More like this request handler -->\n  <requestHandler name="/mlt" class="solr.MoreLikeThisHandler" />\n\n\n  <!-- A Robust Example|'g solr/*/conf/solrconfig.xml
 
-# Add MoreLikeThis handler
-perl -p -i -e 's|<!-- A Robust Example|<!-- More like this request handler -->\n  <requestHandler name="/mlt" class="solr.MoreLikeThisHandler" />\n\n\n  <!-- A Robust Example|'g solr/*/conf/solrconfig.xml
+    echo 'Starting server'
+    # We use exec to allow process monitors to correctly kill the
+    # actual Java process rather than this launcher script:
+    export CMD="java -Djetty.port=9001 -Djava.awt.headless=true -Dapple.awt.UIElement=true -jar start.jar"
 
-echo 'Starting server'
-# We use exec to allow process monitors to correctly kill the
-# actual Java process rather than this launcher script:
-export CMD="java -Djetty.port=9001 -Djava.awt.headless=true -Dapple.awt.UIElement=true -jar start.jar"
+    if [ -z "${BACKGROUND_SOLR}" ]; then
+        exec $CMD
+    else
+        exec $CMD >/dev/null &
+    fi
 
-if [ -z "${BACKGROUND_SOLR}" ]; then
-    exec $CMD
 else
-    exec $CMD >/dev/null &
+    if [[ ${SOLR_VERSION} == 5.* ]]; then
+
+        echo "Extracting Solr ${SOLR_VERSION} to `pwd`/solr-${SOLR_VERSION}/"
+        tar -xf ${SOLR_ARCHIVE}
+        echo "cd into solr 5 directory"
+        cd solr-${SOLR_VERSION}
```

Comment:
This tries to cd into `solr-5.5.0` when the unzipped folder is `solr-5.5.3`. Since line 66 ensures we're on version 5, we might as well change this to `cd solr-5.*`